frame/alliance/src/lib.rs

```diff
@@ -765,6 +702,60 @@ pub mod pallet {
 			Ok(())
 		}
 
+		/// Disband the Alliance, remove all active members and unreserve deposits.
+		///
+		/// Witness data must be set.
+		#[pallet::weight(T::WeightInfo::disband(
+			witness.voting_members,
+			witness.ally_members,
+			witness.voting_members + witness.ally_members,
+		))]
+		pub fn disband(
+			origin: OriginFor<T>,
+			witness: DisbandWitness,
+		) -> DispatchResultWithPostInfo {
+			ensure_root(origin)?;
+
+			ensure!(!witness.is_zero(), Error::<T, I>::BadWitness);
+			ensure!(
+				Self::voting_members_count() <= witness.voting_members,
+				Error::<T, I>::BadWitness
+			);
+			ensure!(Self::ally_members_count() <= witness.ally_members, Error::<T, I>::BadWitness);
+			ensure!(Self::is_initialized(), Error::<T, I>::AllianceNotYetInitialized);
+
+			let mut voting_members = Self::voting_members();
+			T::MembershipChanged::change_members_sorted(&[], &voting_members, &[]);
+
+			let mut ally_members = Self::members_of(MemberRole::Ally);
+			let mut unreserve_count: u32 = 0;
+			for member in voting_members.iter().chain(ally_members.iter()) {
+				if let Some(deposit) = DepositOf::<T, I>::take(&member) {
+					let err_amount = T::Currency::unreserve(&member, deposit);
+					debug_assert!(err_amount.is_zero());
+					unreserve_count += 1;
+				}
+			}
+
+			Members::<T, I>::remove(&MemberRole::Founder);
+			Members::<T, I>::remove(&MemberRole::Fellow);
+			Members::<T, I>::remove(&MemberRole::Ally);
+
+			let voting_members_count = voting_members.len() as u32;
+			let ally_members_count = ally_members.len() as u32;
+
+			voting_members.sort();
+			ally_members.sort();
+			Self::deposit_event(Event::AllianceDisbanded { voting_members, ally_members });
```

Comment:
These could potentially be long lists. I'm not sure if this has weight implications with writing the event ( @ggwpez ?).

Reply:
Since we do not see the particular use case for the list of ids, I am replacing it by counts, the user check the counts.